### ISSUE TITLE
Not able to run a job at a specific fomart

### ISSUE THREAD

**Issue by zzjin** [ondrejbartas/sidekiq-cron#128]
when using sidekiq with sidekiq-cron, we found wired action about some corn jobs.
When corns are set as `0 * * * *`(run at first minute per hour) every thing go right, but set as `0 16 * * *`(run at 16:00 per day) jobs not going to performed.

sidekiq: 4.2.7 sidekiq-cron 0.4.5

when rolling back to sidekiq 4.1.0 with sideki-corn 0.4.2 everything goes right.
may be related to #94  #89 but we only get day jobs run error and we are sepc the 5 conrtab right.
sidekiq log show that cron jobs added,but when get time, no day job added

**Comment by zzjin** [comment]
update:
found where error exists.
same using `sidekiq 4.2.7 with sidekiq-cron 0.4.5`
when using rufus-scheduler: 3.2.0 everything go right, but when using rufus-scheduler 3.3.1(which default when run `bundle install`) issue happened.

**Comment by jmettraux** [comment]
Hello,

what happens when you schedule at `* * * * *` with rufus-scheduler 3.3.1?

**Comment by zzjin** [comment]
@jmettraux 
linux version: ubuntu 16.04.1
ruby version: 2.3.1p112

And yes, `* * * * *` not work too.

**Comment by jmettraux** [comment]
@zzjin it's weird that `0 * * * *` works then.

**Comment by zzjin** [comment]
Sorry, after fully restart sidekiq `* * * * *` worked fine.

**Comment by zzjin** [comment]
update to pre-relase rufus-scheduler 3.3.2 still not work

**Comment by jmettraux** [comment]
@zzjin and you just said that `* * * * *` worked fine with 3.3.1. You are difficult to follow.

**Comment by zzjin** [comment]
@jmettraux What I mean is: after update to 3.3.2,`0 * * * *` AND`* * * * *` works fine, but `0 16 * * *` still not work...
so taht 3.3.X always behave the same and `0 16 * * *` get broken.

**Comment by jmettraux** [comment]
What does running `ruby -e "p [ ENV['TZ'], Time.now.zone ]"` on your system yield? Thanks in advance.

**Comment by zzjin** [comment]
`ruby -e "p [ ENV['TZ'], Time.now.zone ]"` => `[nil, "CST"]`
A few hours ago I also think it about timezone error, so I test with 720 workers that run each 2 min with corn like `2 1 * * *` OR `4 9 * * *` . But no one works.

**Comment by jmettraux** [comment]
@zzjin and what does `bundle exec ruby -e "require 'rufus-scheduler'; require 'pp'; pp [ '* * * * *', '0 * * * *', '0 16 * * *' ].collect { |s| Rufus::Scheduler::CronLine.new(s).to_a }"` yield? Thanks in advance.

**Comment by zzjin** [comment]
It's wired.
when run at 3.3.1(sidekiq-cron: 4.2.7):
```
[[[0], nil, nil, nil, nil, nil, nil, "America/Bahia_Banderas"],
 [[0], [0], nil, nil, nil, nil, nil, "America/Bahia_Banderas"],
 [[0], [0], [16], nil, nil, nil, nil, "America/Bahia_Banderas"]]
```

**Comment by jmettraux** [comment]
Indeed, this is what I see after setting the timezone to "Asia/Shanghai":
```
205610 satsuma 我 ~/w/rufus-scheduler (master) マ TZ="Asia/Shanghai" bundle exec ruby -e "require 'rufus-scheduler'; require 'pp'; pp [ '* * * * *', '0 * * * *', '0 16 * * *' ].collect { |s| Rufus::Scheduler::CronLine.new(s).to_a }"
[[[0], nil, nil, nil, nil, nil, nil, "America/Bahia_Banderas"],
 [[0], [0], nil, nil, nil, nil, nil, "America/Bahia_Banderas"],
 [[0], [0], [16], nil, nil, nil, nil, "America/Bahia_Banderas"]]
```

I recommend you stick with rufus-scheduler 3.2.x until I fix rufus-scheduler 3.3.2.

@ondrejbartas sorry for the noise, not a sidekiq-cron issue.

**Comment by zzjin** [comment]
Got it. I'll use 3.2.X instead. sorry for ondrejbartas too.
